Add recipe for bufferfile

jamescherti opened the pull request:
### Brief summary of what the package does

This package provides helper functions to delete and rename buffer files:
- `(bufferwizard-rename)`: Renames the file that the current buffer is visiting. This command renames the file name on disk, adjusts the buffer name, and updates any indirect buffers or other buffers associated with the old file.
- `(bufferwizard-delete)`: Delete the file associated with a buffer and kill all buffers visiting the file, including indirect buffers.

### Direct link to the package repository

https://github.com/jamescherti/bufferfile.el

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  I noticed you've got other PR(s) already open #9415.  To split my review time evenly between authors, I try to do one PR at a time per author, but I promise I'll get to this when we resolve earlier PRs.

riscy commented:
Thanks again.  Let me know what you think, but I was considering your package summary:
```
Delete or rename buffer file names efficiently 
```
And I thought this might fit better:
```
Rename or delete buffers and associated files 
```
My reason being I feel that `buffer-file-name` is a specific thing in Emacs, and *deleting* a buffer file name feels unusual.

I also recommend renaming the functions from `bufferfile-rename-file` to `bufferfile-rename`, and `bufferfile-delete-file` to `bufferfile-delete` - after all you're renaming and deleting "buffer/file" pairs, not just the file.

I have one more suggestion which is that you add a function `bufferfile-copy`.  This would obviously just be a wrapper around the builtin `write-file` but the nice thing about it is that you present users with a uniform interface around the usual acts of renaming, copying, and deleting.

Let me know what you think about these -

riscy commented:
(If you added a `copy` function the summary line would change too)

jamescherti commented:
Hello @riscy,

> Thanks again. Let me know what you think, but I was considering your package summary:
> 
> ```
> Delete or rename buffer file names efficiently 
> ```
> 
> And I thought this might fit better:
> 
> ```
> Rename or delete buffers and associated files 
> ```
> My reason being I feel that `buffer-file-name` is a specific thing in Emacs, and _deleting_ a buffer file name feels unusual.

The main benefit of this package is the ability to rename or delete buffer file names along with their associated buffers (e.g., clones/indirect buffers).

I have changed it to: Rename or Delete Buffers and Associated Files and Buffers

> I also recommend renaming the functions from `bufferfile-rename-file` to `bufferfile-rename`, and `bufferfile-delete-file` to `bufferfile-delete` - after all you're renaming and deleting "buffer/file" pairs, not just the file.

Thank you for pointing this out. The README.md was outdated.  The function names are actually `bufferfile-rename` and `bufferfile-delete`.

> I have one more suggestion which is that you add a function `bufferfile-copy`. This would obviously just be a wrapper around the builtin `write-file` but the nice thing about it is that you present users with a uniform interface around the usual acts of renaming, copying, and deleting.

Thank you for your suggestion. I have added it `bufferfile-copy` to my queue, and it will be included in the future.
 
> Let me know what you think about these -

riscy commented:
Thank you!  Just a reminder I send everyone:

- In case you haven't, ensure GitHub release 1.0.3 is up-to-date with your current code and `Package-Version`
